subways/tests/test_route.py

```diff
@@ -0,0 +1,126 @@
+from unittest import TestCase
+
+from subways.structure.route import (
+    get_interval_in_seconds_from_tags,
+    osm_interval_to_seconds,
+    parse_time_range,
+)
+
+
+class TestTimeIntervalsParsing(TestCase):
+    def test__osm_interval_to_seconds__invalid_value(self) -> None:
+        intervals = (
+            ["", "0", "abc", "x30", "30x", "3x0"]
+            + ["5:", ":5", "01:05:", ":01:05", "01:01:00:", ":01:01:00"]
+            + ["01x:05", "01:x5", "x5:01:00", "01:0x:00", "01:01:x"]
+            + ["-5", "01:-05", "-01:05", "-01:00:00", "01:-01:00", "01:01:-01"]
+        )
+        for interval in intervals:
+            with self.subTest(msg=f"value='{interval}'"):
+                self.assertIsNone(osm_interval_to_seconds(interval))
+
+    def test__osm_interval_to_seconds__valid_value(self) -> None:
+        intervals = ["5", "65", "01:00", "10:55", "02:02:02", "2:2:2"]
```

Comment:
is something like "00:25"/"00:51" (interval with 0 hours and non-0 minutes) possible?

Also "00:00" or "00:61"?

Reply:
A fix and tests added, thanks.